anyway/infographics_utils.py

```diff
@@ -169,16 +178,21 @@ def get() -> List[Type[Widget]]:
     @staticmethod
     def register(widget_class: Type[Widget]) -> Type[Widget]:
         WidgetCollection.widgets.append(widget_class)
+        WidgetCollection.widgets_dict[widget_class.widget_id.name] = widget_class
+        logging.debug(f"register:{widget_class.widget_id.name}:{widget_class}\n")
         return widget_class
 
 
 @WidgetCollection.register
 class AccidentCountBySeverityWidget(Widget):
-    widget_id: WidgetId = dataclasses.field(init=False, default=WidgetId.accident_count_by_severity)
+    widget_id: WidgetId = WidgetId.accident_count_by_severity
```

Comment:
I don't see the need for `WidgeId` anymore. The enum is effectively replaced by the widget-collection. `widget_id` is simply the name of the class. Nothing needs to be explicitly initialized or passed around.

Reply:
and `name` is essentially a nicer way of saying `cls.__name__`. If you want this syntactic sugar, it can be defined once, as a trivial classmethod.

Reply:
The names we use are lowercase with underscore word separators. Class names are Camel-case. How do you suggest to overcome this discrepancy when using `cls.__name__`?

Reply:
Where do we use these names Where are the created? If they originate inside the code, they can be changed; if they are not, and we rely on the to reflect variable names inside the program (the names of the enum instances) then we expose implementation detail to the outside.

I thought they come from iterating over `WidgetCollection.widgets`.

Reply:
The name is a field in the widgets that the api returns. I assume its goal is to identify the widget in the UI. We can change the current names to class names (camel notation). I do not know if and to what extent it affects the UI.

Reply:
Regardless of whether this is camel case or snake case, it sounds like relying on implementation detail (name of the enum values, name of the class) in the API. It's not horrible but it's not ideal either.

Reply:
I am setting the enum's name to the string I need. Is it different from setting a constant to the value I need?

Reply:
Yes, because the enum's name is internal to the program, and referring to it is a kind of metaprogramming. Again - this is not horrible, it's just something to keep in mind.

Reply:
@elazarg 
We can do 
`class WidgetId(Enum):`
`         accident_count_by_severity = 'accident_count_by_severity'`
`         most_severe_accidents_table = 'most_severe_accidents_table'`

and then
`WidgetId.accident_count_by_severity.value`
This way it does not rely on implementation. Do you prefer it?

Reply:
I think it's better, yes. We can equivalently do
```python
@WidgetCollection.register('accident_count_by_severity')
class AccidentCountBySeverityWidget(Widget):
    ...
```

Reply:
I see you prefer enums, It's fine, It's just that the enum represents not a widget instance but a widget type, and we already have exactly that when we define a class, so it feels like redundancy to me.

Reply:
The benefit in having enum is that the reader can see a list of available widgets.

Reply:
I think I finally understand. I removed the `WidgetId` `Enum`.